portal/views/home_new.py

```diff
@@ -0,0 +1,179 @@
+# authorisation of Ocado.
+#
+# Any propagation, distribution or conveyance of this program must include this
+# copyright notice and these terms. You must not misrepresent the origins of this
+# program; modified versions of the program must be marked as such and not
+# identified as the original program.
+from functools import partial
+
+from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import reverse, reverse_lazy
+from django.contrib import messages as messages
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.forms import AuthenticationForm
+from recaptcha import RecaptchaClient
+from django_recaptcha_field import create_form_subclass_with_recaptcha
+
+from portal.forms.teach import TeacherLoginForm
+from portal.helpers.emails import send_verification_email, is_verified
+from portal.utils import using_two_factor
+from portal import app_settings
+from ratelimit.decorators import ratelimit
+
+recaptcha_client = RecaptchaClient(app_settings.RECAPTCHA_PRIVATE_KEY, app_settings.RECAPTCHA_PUBLIC_KEY)
+
+
+def teach_email_labeller(request):
+    if request.method == 'POST' and 'login' in request.POST:
+        return request.POST['login-email']
+
+    return ''
+
+
+def login_popup(request):
+    return render_forms(request, 'redesign/login_popup.html')
+
+
+def home_new(request):
+    return render_forms(request, 'redesign/home_new.html')
+
+
+def teach_new(request):
+    return render_forms(request, 'redesign/teach_new.html')
+
+
+def play_new(request):
+    return render_forms(request, 'redesign/play_new.html')
+
+
+def about_new(request):
+    return render_forms(request, 'redesign/about_new.html')
+
+
+def help_and_support_new(request):
+    return render_forms(request, 'redesign/help-and-support_new.html')
+
+
+def terms_new(request):
+    return render_forms(request, 'redesign/terms_new.html')
+
+
+@ratelimit('ip', periods=['1m'], increment=lambda req, res: hasattr(res, 'count') and res.count)
+@ratelimit('email', labeller=teach_email_labeller, ip=False, periods=['1m'], increment=lambda req, res: hasattr(res, 'count') and res.count)
+def render_forms(request, rendered_url):
+    invalid_form = False
+
+    limits = getattr(request, 'limits', {'ip': [0], 'email': [0]})
+    captcha_limit = 5
+
+    LoginFormWithCaptcha = partial(create_form_subclass_with_recaptcha(TeacherLoginForm, recaptcha_client), request)
+    InputLoginForm = compute_input_login_form(LoginFormWithCaptcha, limits, captcha_limit)
+    OutputLoginForm = compute_output_login_form(LoginFormWithCaptcha, limits, captcha_limit)
+
+    login_form = OutputLoginForm(prefix='login')
+
+    if request.method == 'POST':
+        if 'login' in request.POST:
+            login_form = InputLoginForm(request.POST, prefix='login')
+            if login_form.is_valid():
+                return process_form(request, login_form)
```

Comment:
if you'll use a different function for the registration form, then you might want to give this function a more specific name than process_form?